pkg/github/github_helper.go

```diff
@@ -304,3 +308,41 @@ func RefineGitHostingServiceError(response *http.Response, originErr error) erro
 func getDefaultBranch(client *GithubClient, owner string, repository string) (string, error) {
 	return client.getDefaultBranch(owner, repository)
 }
+
+// findUnmergedOnboardingMergeRequest finds out the unmerged merge request that is opened during the component onboarding
+// An onboarding merge request fulfills both:
+// 1) opened based on the base branch which is determined by the Revision or is the default branch of component repository
+// 2) opened from head ref: owner:appstudio-{component.Name}
+// If no onboarding merge request is found, nil is returned.
+func findUnmergedOnboardingMergeRequest(
+	ghclient *GithubClient, owner, repository, sourceBranch, baseBranch, authorName string) (*github.PullRequest, error) {
+	opts := &github.PullRequestListOptions{
+		Head: fmt.Sprintf("%s:%s", authorName, sourceBranch),
+		Base: baseBranch,
+		// Opened pull request is searched by default by GitHub API.
+	}
+	pullRequests, resp, err := ghclient.client.PullRequests.List(context.Background(), owner, repository, opts)
+	if err != nil {
+		return nil, RefineGitHostingServiceError(resp.Response, err)
+	}
+	if len(pullRequests) == 0 {
+		return nil, nil
+	}
+	return pullRequests[0], nil
+}
+
+func closeMergeRequest(ghclient *GithubClient, owner, repository string, pullRequest *github.PullRequest) (*github.PullRequest, error) {
+	updateData := &github.PullRequest{
+		Title:               pullRequest.Title,
+		Body:                pullRequest.Body,
+		State:               github.String("closed"),
+		MaintainerCanModify: pullRequest.MaintainerCanModify,
+	}
+	pullRequest, resp, err := ghclient.client.PullRequests.Edit(
+		context.Background(), owner, repository, pullRequest.GetNumber(), updateData)
+	return pullRequest, RefineGitHostingServiceError(resp.Response, err)
+}
+
+func deleteBranch(client *GithubClient, owner, repository, branch string) error {
+	return client.deleteBranch(owner, repository, branch)
```

Comment:
re-use `return client.deleteReference(owner, repository, branch)` then it works properly. You can delete deleteBranch function from github_client.go.